bundles/org.openhab.binding.etherrain/src/main/java/org/openhab/binding/etherrain/internal/handler/EtherRainHandler.java

```diff
@@ -0,0 +1,213 @@
+        this.updateJob = scheduler.scheduleWithFixedDelay(this::updateBridge, 0, config.refresh, TimeUnit.SECONDS);
+
+        logger.debug("EtherRain sucessfully initialized. Starting status poll at: {}", config.refresh);
+    }
+
+    private void stopUpdateJob() {
+        logger.debug("Stopping Etherrain Update Job");
+
+        final ScheduledFuture<?> updateJob = this.updateJob;
+        if (updateJob != null && !updateJob.isDone()) {
+            updateJob.cancel(false);
+        }
+
+        this.updateJob = null;
+    }
+
+    @SuppressWarnings("null")
+    private boolean updateBridge() {
+        if (!connected || device == null) {
+            connected = connectBridge();
+            if (!connected || device == null) {
+                connected = false;
+                device = null;
+                logger.debug("Could not connect to Etherrain device.");
+                return false;
+            }
+        }
+
+        EtherRainStatusResponse response;
+
+        try {
+            response = device.commandStatus();
+        } catch (EtherRainException | IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                    "Could not create a connection to the EtherRain");
+            logger.debug("Could not open API connection to the EtherRain device. Exception received: {}",
+                    e.getMessage());
+            device = null;
+            return false;
+        }
+
+        updateState(EtherRainBindingConstants.CHANNEL_ID_OPERATING_STATUS,
+                new StringType(response.getOperatingStatus().name()));
+
+        updateState(EtherRainBindingConstants.CHANNEL_ID_COMMAND_STATUS,
+                new StringType(response.getLastCommandStatus().name()));
+
+        switch (response.getLastCommandResult()) {
+            case OK:
+                updateState(EtherRainBindingConstants.CHANNEL_ID_OPERATING_RESULT, new StringType("OK"));
+                break;
+            case RN:
+                updateState(EtherRainBindingConstants.CHANNEL_ID_OPERATING_RESULT, new StringType("RAIN INTERRUPTED"));
+                break;
+            case SH:
+                updateState(EtherRainBindingConstants.CHANNEL_ID_OPERATING_RESULT,
+                        new StringType("INTERRUPPTED SHORT"));
+                break;
+            case NC:
+                updateState(EtherRainBindingConstants.CHANNEL_ID_OPERATING_RESULT, new StringType("DID NOT COMPLETE"));
+                break;
+        }
+
+        updateState(EtherRainBindingConstants.CHANNEL_ID_RELAY_INDEX, new DecimalType(response.getLastActiveValue()));
+
+        OnOffType rs = OnOffType.from(response.isRainSensor());
+
+        updateState(EtherRainBindingConstants.CHANNEL_ID_SENSOR_RAIN, rs);
+
+        logger.debug("Completed Etherrain Update");
+
+        return true;
+    }
+
+    private boolean execute() {
+        if (device != null) {
```

Comment:
This helps deals with the null warnings.
```suggestion
        EtherRainCommunication device = this.device;
        if (device != null) {
```